networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/DiscoveryNetwork.java

```diff
@@ -164,10 +169,21 @@ public void setForkInfo(final ForkInfo currentForkInfo, final Optional<Fork> nex
     // If no future fork is planned, set next_fork_epoch = FAR_FUTURE_EPOCH to signal this
     final UnsignedLong nextForkEpoch = nextForkInfo.map(Fork::getEpoch).orElse(FAR_FUTURE_EPOCH);
 
-    final EnrForkId enrForkId =
-        new EnrForkId(currentForkInfo.getForkDigest(), nextVersion, nextForkEpoch);
-    discoveryService.updateCustomENRField(
-        ETH2_ENR_FIELD, SimpleOffsetSerializer.serialize(enrForkId));
+    final Bytes4 forkDigest = currentForkInfo.getForkDigest();
+    final EnrForkId enrForkId = new EnrForkId(forkDigest, nextVersion, nextForkEpoch);
+    final Bytes encodedEnrForkId = SimpleOffsetSerializer.serialize(enrForkId);
+
+    discoveryService.updateCustomENRField(ETH2_ENR_FIELD, encodedEnrForkId);
+
+    long newPeerPredicateId =
+        connectionManager.addPeerPredicate(
+            peer ->
+                peer.getEnrForkId().equals(encodedEnrForkId)
```

Comment:
What happens when the peer doesn't include an `eth2` field in it's ENR? I think we should not connect to them, but I think we'll get `NullPointerException`.

Reply:
Fixed.

Reply:
Fixed this.